ci/pipeline.yml

```diff
@@ -0,0 +1,305 @@
+---
+staging-cf: &staging-cf
+  CF_API: ((staging-cf-api))
+  CF_USERNAME: ((staging-cf-username))
+  CF_PASSWORD: ((staging-cf-password))
+  CF_ORG: ((staging-cf-org))
+  CF_SPACE: ((staging-cf-space))
+  MIRROR_HOSTNAME: ((staging-mirror-hostname))
+
+production-cf: &production-cf
+  CF_API: ((production-cf-api))
+  CF_USERNAME: ((production-cf-username))
+  CF_PASSWORD: ((production-cf-password))
+  CF_ORG: ((production-cf-org))
+  CF_SPACE: ((production-cf-space))
+  MIRROR_HOSTNAME: ((production-mirror-hostname))
+
+jobs:
+
+- name: configure-pipeline
+  serial_groups:
+  - mirror
+  - proxy
+  plan:
+  - in_parallel:
+    - get: pipeline-config
+      trigger: true
+    - get: pipeline-source
+      trigger: true
+      params: {depth: 1}
+  - set_pipeline: docker-registry-mirror
+    file: pipeline-source/ci/pipeline.yml
+    var_files: 
+      - pipeline-config/docker-registry-mirror.yml
+
+- name: deploy-registry-staging
+  serial_groups: [mirror]
+  plan:
+  - in_parallel: 
+    - get: registry-source
+      trigger: true
+      params: {depth: 1}
+    - get: docker-registry-image
+      trigger: true
+    - get: ci-source
+      params: {depth: 1}
+  - task: create-s3-bucket
+    file: ci-source/ci/tasks/create-s3-bucket.yml
+    params:
+      <<: *staging-cf
+  - task: deploy-registry
+    file: ci-source/ci/tasks/deploy-registry.yml
+    params:
+      <<: *staging-cf
+    # Don't cancel the deployment in staging so we can debug the failure
+    # on_failure:
+    #   try:
+    #     task: cancel-deployment
+    #     file: ci-source/ci/tasks/cancel-deployment.yml
+    #     params:
+    #       <<: *staging-cf
+    #       CF_APP_NAME: docker-registry-mirror
```

Comment:
Thanks for keeping this config here (and elsewhere) with an explicit note!  I think that will help with future maintenance of this in the long run.  I know it's also documented in the README (which is also good!), but calling it out here specifically where the configuration is helps with understanding and context.